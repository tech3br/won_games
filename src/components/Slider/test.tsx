import { render, screen } from '@testing-library/react'
import 'match-media-mock'
import { Settings } from 'react-slick'
import styled from 'styled-components'
import { renderWithTheme } from 'utils/tests/helpers'

import Slider from '.'

const settings: Settings = {
    dots: true,
    infinite: true,
    speed: 500,
    slidesToShow: 3,
    slidesToScroll: 1
}

const verticalSettings: Settings = {
    vertical: true,
    verticalSwiping: true,
    dots: true,
    infinite: false,
    slidesToShow: 1
}

const Slide = styled.div`
    background-color: gray;
    width: 30rem;
    padding: 10rem 0;
    border: 0.1rem solid red;
    color: white;
    text-align: center;
`

describe('<Slider />', () => {
    it('should render horizontal correctly', () => {
        renderWithTheme(
            <Slider settings={settings}>
                <Slide>1</Slide>
                <Slide>2</Slide>
                <Slide>3</Slide>
                <Slide>4</Slide>
                <Slide>5</Slide>
                <Slide>6</Slide>
            </Slider>
        )
    })

    it('should render vertical correctly', () => {
        renderWithTheme(
            <Slider settings={verticalSettings}>
                <Slide>1</Slide>
                <Slide>2</Slide>
                <Slide>3</Slide>
                <Slide>4</Slide>
                <Slide>5</Slide>
                <Slide>6</Slide>
            </Slider>
        )
    })

    it('should render children as slider item', () => {
        const { container } = render(
            <Slider settings={{ slidesToShow: 1, infinite: false }}>
                <p>item 1</p>
                <p>item 2</p>
            </Slider>
        )

        expect(
            screen.getByText(/item 1/i).parentElement?.parentElement
        ).toHaveClass('slick-slide')

        expect(
            screen.getByText(/item 2/i).parentElement?.parentElement
        ).toHaveClass('slick-slide')

        expect(container.firstChild).toMatchSnapshot()
    })
})
