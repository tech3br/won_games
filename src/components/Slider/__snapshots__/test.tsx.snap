// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> should render children as slider item 1`] = `
<section
  class="styles__Wrapper-sc-24vst-0 hZbfrR"
>
  <div
    class="slick-slider slick-initialized"
    dir="ltr"
  >
    <button
      class="slick-arrow slick-prev slick-disabled"
      data-role="none"
      style="display: block;"
      type="button"
    >
       
      Previous
    </button>
    <div
      class="slick-list"
    >
      <div
        class="slick-track"
        style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
      >
        <div
          aria-hidden="false"
          class="slick-slide slick-active slick-current"
          data-index="0"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <p
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              item 1
            </p>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide"
          data-index="1"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <p
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              item 2
            </p>
          </div>
        </div>
      </div>
    </div>
    <button
      class="slick-arrow slick-next"
      data-role="none"
      style="display: block;"
      type="button"
    >
       
      Next
    </button>
  </div>
</section>
`
